grails-app/services/org/pih/warehouse/product/ProductMergeService.groovy

```diff
@@ -365,6 +359,19 @@ class ProductMergeService {
         // RefreshOrderSummaryJob.triggerNow() // <- FIXME: maybe do a refresh by list of affected orders instead of full table
     }
 
+    /**
+     * Finds inventory item from primary product, that has the same lot and expiry date as obsoleteInventoryItem
+     * (null lotNumber and empty string lotNumber are treated equally)
+     * */
+    def getPrimaryInventoryItem(Set<InventoryItem> primaryInventoryItems, InventoryItem obsoleteInventoryItem) {
+        def obsoleteLotNumber = obsoleteInventoryItem?.lotNumber
+        def obsoleteExpirationDate = obsoleteInventoryItem?.expirationDate
+        return primaryInventoryItems?.find {
+            def exactLotNumber = obsoleteLotNumber ? (it.lotNumber == obsoleteLotNumber) : (it.lotNumber == null || it.lotNumber == "")
```

Comment:
Don't forget about the whitespace case i.e. if someone enters more than one space character or worse ... tabs, returns, etc.

For readability, it might be better to move this method to Product.